### DISCUSSION TITLE
Face shape detection (random idea)

### DISCUSSION THREAD

**Discussion by uzair004** [vladmandic/human#188] [Ideas]
Just randomly got a nice idea for "Face shape detection" e.g Diamond, round, square... (Total 7)
It involves two measurements from top to chin, nose to both ears.
I guess there will be a model for doing such things. Adding support to human will be nice addition.

P.S: not desparetly looking for this feature or anything, just random idea I'll would try to create my next NodeJS project with.

**Comment by vladmandic** [comment]
Should be doable - if you provide some definitions of shapes :)
E.g. what is *diamond* vs *round* vs *square* (just don't say it *looks-like* a shape) :)

**Reply by uzair004** [reply]
So human face  are dividend into 7 shapes
The 7 basic face shapes are oval, round, square, diamond, heart, pear and oblong.

It is determined based on face calculation to which category certain shape belong.
It involves 4 calculations
1. Forehead width (at widest point i.e center)
2. Cheekbone width
3. Jawline width
4. Face height (from chin to top hairline)

More details can be found here [Determine face shape in 3 steps](https://www.kraywoods.com/blogs/our-stories/face-shape-guide) but basically it involves which measurements among above 4 is highest that determine face shape e.g if face height is more then width it's Oval shape, in case of opposite measurements it's round or square. Adding cheekbone measurements make it more specific which face shape it could be.

In manual method it includes measuring from hairline to another, chin to top hairline, widest point of forehead (center), there will be more specific guide, but this is what i have found as of now mostly used.

**Comment by vladmandic** [comment]
Article is pretty clear and all of the measurements are already available, just need to measure distance between set points and then implement rules. All of the face points are defined here <https://github.com/vladmandic/human/blob/main/assets/facemesh.png>

So forehead width would be between points `face.mesh[63]` to `face.mesh[293]` (pick any other points if you think they fit what is forehead better) and then measure distance between them: `sqrt((x1 - x0)^2 + (y1 - y0)^2)` (not just 1d x-distance since head can be under angle so its beter to look at 2d distance).

**If you care to contribute, this would be a good start! :)**

Once you have measurements for everything needed in the article, just run through rules and determine shape.

**Comment by uzair004** [comment]
> https://github.com/vladmandic/human/blob/main/assets/facemesh.png

Image contain nice visualization of face points, I'll give a try once I am done with my FYP :)

**Comment by uzair004** [comment]
> https://github.com/vladmandic/human/blob/main/assets/facemesh.png

Image contain nice visualization of face points, I'll give a try once I am done with my FYP :)

**Comment by peipegzzz** [comment]
Give this face shape detector a shot to figure out your face shape. It's a cool AI tech that not only IDs your face shape but also throws in some other interesting info. 
https://www.perfectcorp.com/business/showcase/face-analyzer

**Reply by wendyyennn31** [reply]
Tried, shows lots of my face features

**Comment by Enaeemullah** [comment]
This already exists in practice, btw. Tools like [https://faceshapedetector.app](https://faceshapedetector.app)
 do face-shape classification (oval, round, square, diamond, etc.) by analysing facial proportions via landmark detection.

Accuracy obviously depends on landmark quality and how you define the categories (the labels themselves are fuzzy), but it’s a decent example of how far you can get without overcomplicating things.

**Comment by Adil512** [comment]
This sounds like a really interesting idea! Using key facial measurements like top-to-chin and nose-to-ear distances can definitely help in accurately classifying face shapes like diamond, round, square, and other types. There are AI models that specialize in this, so integrating one could be a great addition to a NodeJS project.

For reference, there are a couple of excellent implementations of face shape detection online that you might find useful for inspiration: **https://whatismyfaceshape.net/**  and **https://faceshapedetector.us/**
. Both provide fast and accurate AI-powered analysis for multiple face shapes.

Looking forward to seeing what you build!

**Comment by vladmandic** [comment]
ok, cool idea, but what can be done with it? lets say we identify face shape - what is that information good for in sdnext? what can we use it for?

**Comment by wendyyennn31-ops** [comment]
I tried many tools on the market. Here is a summary of the ones I've tried:

1. [https://www.perfectcorp.com/business/showcase/face-analyzer](https://www.perfectcorp.com/business/showcase/face-analyzer) 
Enterprise-level accuracy and used by international brands, this tool detects face shapes (like round, square, diamond and others). It also provides a comprehensive analysis of lip shape, eyebrow shape, and personal colors etc etc. 

2. [https://faceshapedetector.app/](https://faceshapedetector.app/)
You need to manually select either male or female first. Then it provides a score based on few basic analysis, such as face width and length. It also offers suggestions for hairstyles and glasses. The difference is it doesn't provide a virtual try on feature like [https://www.perfectcorp.com/business/showcase/face-analyzer](https://www.perfectcorp.com/business/showcase/face-analyzer).

3. [https://whatismyfaceshape.net/](https://whatismyfaceshape.net/)
First thought is too much ads on this website.
The measuring results are the simliest i've used, presenting face shapes, and providing one sentence to tell you your suitable hairstyles etc.
